paperqa/llms.py

```diff
@@ -824,16 +887,65 @@ async def similarity_search(
 
 
 def embedding_model_factory(embedding: str, **kwargs) -> EmbeddingModel:
-    if embedding.startswith("hybrid"):
-        embedding_model_name = "-".join(embedding.split("-")[1:])
-        return HybridEmbeddingModel(
-            models=[
-                LiteLLMEmbeddingModel(name=embedding_model_name),
-                SparseEmbeddingModel(**kwargs),
-            ]
+    """
+    Factory function to create an appropriate EmbeddingModel based on the embedding string.
+
+    Supports:
+    - SentenceTransformer models prefixed with "st-" (e.g., "st-multi-qa-MiniLM-L6-cos-v1")
+    - LiteLLM models (default if no prefix is provided)
+    - Hybrid embeddings prefixed with "hybrid-", contains a sparse and a dense model
+
+    Args:
+        embedding: The embedding model identifier. Supports prefixes like "st-" for SentenceTransformer
+                         and "hybrid-" for combining multiple embedding models.
+        **kwargs: Additional keyword arguments for the embedding model.
+
+    Returns:
+        EmbeddingModel: An instance of a subclass of EmbeddingModel.
+    """
+    embedding = embedding.strip()  # Remove any leading/trailing whitespace
+
+    if embedding.startswith("hybrid-"):
+        # Extract the component embedding identifiers after "hybrid-"
+        dense_name = embedding[len("hybrid-") :]
+
+        if not dense_name:
+            raise ValueError(
+                "Hybrid embedding must contain at least one component embedding."
+            )
+
+        # Recursively create each component embedding model
+        dense_model = embedding_model_factory(dense_name, **kwargs)
+        sparse_model = SparseEmbeddingModel(**kwargs)
+
+        return HybridEmbeddingModel(models=[dense_model, sparse_model])
+
+    if embedding.startswith("st-"):
+        # Extract the SentenceTransformer model name after "st-"
+        model_name = embedding[len("st-") :].strip()
+        if not model_name:
+            raise ValueError(
+                "SentenceTransformer model name must be specified after 'st-'."
+            )
+
+        return SentenceTransformerEmbeddingModel(
+            name=model_name,
+            config=kwargs,  # Pass any additional configurations via config
+        )
+
+    if embedding.startswith("litellm-"):
+        # Extract the LiteLLM model name after "litellm-"
+        model_name = embedding[len("litellm-") :].strip()
+        if not model_name:
+            raise ValueError("LiteLLM model name must be specified after 'litellm-'.")
+
+        return LiteLLMEmbeddingModel(
+            name=model_name,
+            config=kwargs,  # Pass any additional configurations via config
```

Comment:
```suggestion
            config=kwargs
```

Is this comment necessary?